packages/browser-destinations/destinations/google-analytics-4-web/src/__tests__/setConfigurationFields.test.ts

```diff
@@ -0,0 +1,177 @@
+    pageView: boolean
+    cookiePath?: string
+  } = {
+    enableConsentMode: false,
+    measurementID: 'G-XXXXXXXXXX',
+    allowAdPersonalizationSignals: false,
+    allowGoogleSignals: false,
+    cookieDomain: 'auto',
+    cookieExpirationInSeconds: 63072000,
+    cookieUpdate: true,
+    pageView: true
+  }
+  beforeEach(async () => {
+    jest.restoreAllMocks()
+
+    const [trackEventPlugin] = await googleAnalytics4Web({
+      ...settings,
+      subscriptions
+    })
+    setConfigurationEvent = trackEventPlugin
+
+    jest.spyOn(destination, 'initialize').mockImplementation(() => {
+      mockGtag = jest.fn()
+      return Promise.resolve(mockGtag)
+    })
+    await trackEventPlugin.load(Context.system(), {} as Analytics)
+  })
+
+  it('should call gtag with the correct config when consent mode is enabled', async () => {
+    settings.enableConsentMode = true
+
+    const [trackEventPlugin] = await googleAnalytics4Web({
+      ...settings,
+      subscriptions
+    })
+    setConfigurationEvent = trackEventPlugin
+    await trackEventPlugin.load(Context.system(), {} as Analytics)
+
+    const context = new Context({
+      event: 'setConfigurationFields',
+      type: 'page',
+      properties: {
+        ads_storage_consent_state: 'granted',
+        analytics_storage_consent_state: 'denied'
+      }
+    })
+
+    setConfigurationEvent.page?.(context)
+
+    expect(mockGtag).toHaveBeenCalledWith('consent', 'update', {
+      ad_storage: 'granted',
+      analytics_storage: 'denied'
+    })
+    expect(mockGtag).toHaveBeenCalledWith('config', 'G-XXXXXXXXXX', {
+      allow_ad_personalization_signals: false,
+      allow_google_signals: false
+    })
+  })
+  it('should call gtag with the correct config when consent mode is disabled and having values in payload', async () => {
+    settings.enableConsentMode = false
+    const [trackEventPlugin] = await googleAnalytics4Web({
+      ...settings,
+      subscriptions
+    })
+    setConfigurationEvent = trackEventPlugin
+    await trackEventPlugin.load(Context.system(), {} as Analytics)
+
+    const context = new Context({
+      event: 'setConfigurationFields',
+      type: 'page',
+      properties: {
+        screen_resolution: '800*600',
+        user_id: 'segment123',
+        page_title: 'User Registration Page',
+        page_referrer: 'Home',
+        language: 'EN',
+        content_group: '/home/login'
+      }
+    })
+
+    setConfigurationEvent.page?.(context)
+    expect(mockGtag).toHaveBeenCalledWith('config', 'G-XXXXXXXXXX', {
+      screen_resolution: '800*600',
+      user_id: 'segment123',
+      page_title: 'User Registration Page',
+      page_referrer: 'Home',
+      language: 'EN',
+      content_group: '/home/login',
+      allow_ad_personalization_signals: false,
+      allow_google_signals: false
+    })
+  })
+  it('should call gtag with the correct config when consent mode is disabled and cookie other then default value', async () => {
```

Comment:
Since the focus of this test is on cookies, can we just rename the test to not include consent?
```suggestion
  it('should configure cookie when cookie path is other then default value', async () => {
```